### PR TITLE
Changed the waitFrames loop

### PR DESCRIPTION
**Pull request by cbcoolbox**
-If you are on the game menus and try to play the combo the tool won't lock up anymore.  it doesn't do anything but exit the play timeline loop.
-If you try to start a combo on the pause screen it won't lock up. It doesn't do anything but exit the play timeline loop.
-If you pause the game mid combo, the combo stops. Which avoids from it hitting buttons through the menu randomly.

I know you are trying to make this fast enough for everyone to use, but if they can't run the game at 60+ FPS consistently the tool won't work and they won't be able to hit some links anyways.

If you think there's too much logic in the waitframes loop let me know.

### CODE REVIEW

File: SF4ComboTrainer/SF4Control.cs [hunk 1]
```diff
@@ -29,6 +39,9 @@ class SF4Control
         [DllImport("user32.dll")]
         public static extern IntPtr GetForegroundWindow();
 
+        // to see if in a match.
+        public bool inMatch;
```
Comment (Necrophagos) [inline SF4ComboTrainer/SF4Control.cs]: add volatile keyword

File: SF4ComboTrainer/Form1.cs [hunk 2]
```diff
@@ -198,15 +198,20 @@ private void playTimeline()
             for (int i = 0; i < TimeLine.Items.Count; i++)
             {
                 TimeLineItem item = (TimeLineItem)TimeLine.Items[i];
-                item.Action(sf4control, chkSendInputs.Checked);
 
-                
+                // if we aren't in a match (defined by being on a menu or pause is selected) the play timeline stops.
+                if (sf4control.inMatch) item.Action(sf4control, chkSendInputs.Checked);
+
+                // i chose to comment this out because it looks nicer when the highlight goes to the bottom.
```
Comment (Necrophagos) [inline SF4ComboTrainer/Form1.cs]: break, show message box and then manually set the highlight to the last item using the threadsafe code below

File: SF4ComboTrainer/SF4Control.cs [hunk 3]
```diff
@@ -59,11 +72,34 @@ public SF4Control(SF4Memory sf4memory)
...
         {
+            // Reset / start the frameTimer which is used to get time between frames.
+            frameTimer.Reset();
+            frameTimer.Start();
+
             int currentFrame = sf4memory.getFrameCount();
             int endFrame = currentFrame + frames;
-            while (currentFrame < endFrame)
+
+            while (currentFrame < endFrame && frameTimer.ElapsedMilliseconds < MIN_TIME_BETWEEN_FRAMES)
```
Comment (Necrophagos) [inline SF4ComboTrainer/SF4Control.cs:82]: this whole thing is awesome, thx!

**Comment by Necrophagos** [comment]
I love it, thx for your work. Strangely the deletion of InputSimulator.xml and pdb is part of the commit, if you could maybe fix this and the stuff I commented on I'll merge the pull request

**Comment by cbcoolbox** [comment]
Fuck, my bad about the other two files. Good catch with the volatile keyword!! I'll be able to change late tonight so about 12 hours.

File: SF4ComboTrainer/Form1.cs (hunk 2 above)
Comment (cbcoolbox) [inline SF4ComboTrainer/Form1.cs]: Do we really want a message box saying anything?

**Pull request merged**